docs/sql/general/lexical-structure.rst

```diff
@@ -28,6 +28,22 @@ String literals are defined as an arbitrary sequence of characters that are
 delimited with single quotes ``'`` as defined in ANSI SQL, for example
 ``'This is a string'``.
 
+In addition, CrateDB supports dollar quoted strings to help avoid escaping
+single quotes within single quoted strings.
+For example, ``'I''m a string'`` can be re-written as
+``$TAG$I'm a string$TAG$``, where the matching pair of ``TAGs`` can be zero or
+more characters.
+
+::
+
+    cr> select 'I''m a string' = $tag1$I'm a string$tag1$;
+    +------+
+    | true |
+    +------+
+    | TRUE |
+    +------+
+    SELECT 1 row in set (... sec)
+
```

Comment:
Maybe add a note that we don't support nesting tags: `$a$......$b$......$b$.....$a$`